src/plugins/ui_core_views/cell_evaluation/compilation_parameters.ts

```diff
@@ -88,29 +88,30 @@ class CompilationParametersBuilder {
           : _t(
               "Function %s expects its parameters to be single values or single cell references, not ranges.",
               functionName.toString()
-            )
-      );
+            ),
+      };
     }
     if (range.invalidSheetName) {
-      throw new Error(_t("Invalid sheet name: %s", range.invalidSheetName));
+      throw { value: ERR_GENERIC, message: _t("Invalid sheet name: %s", range.invalidSheetName) };
     }
     const position = { sheetId: range.sheetId, col: range.zone.left, row: range.zone.top };
     return this.readCell(position);
   }
 
-  private readCell(position: CellPosition): ValueAndFormat {
+  private readCell(position: CellPosition): UnitData {
     if (!this.getters.tryGetSheet(position.sheetId)) {
-      throw new Error(_t("Invalid sheet name"));
+      throw { value: ERR_GENERIC, message: _t("Invalid sheet name: %s") };
     }
     const evaluatedCell = this.getEvaluatedCellIfNotEmpty(position);
     if (evaluatedCell === undefined) {
       return { value: null, format: this.getters.getCell(position)?.format };
     }
-    return evaluatedCell;
+    const { value, format, message } = evaluatedCell;
+    return { value, format, message };
```

Comment:
Let's avoid creating useless objects.
```suggestion
    return evaluatedCell
```